nomad/resource_job.go

```diff
@@ -629,6 +648,29 @@ func resourceJobRead(d *schema.ResourceData, meta interface{}) error {
 		d.Set("allocation_ids", nil)
 	}
 
+	// Update jobspec submission data if available.
+	// Safely ignore errors as this is an optional step.
+	sub, _, err := client.Jobs().Submission(*job.ID, int(*job.Version), opts)
+	if err != nil {
+		log.Printf("[WARN] failed to read job submission: %v", err)
+	} else if sub != nil {
+		if sub.Source != "" {
+			d.Set("jobspec", sub.Source)
+		}
+
+		// Update HCL2 variables if present.
+		hcl2, ok := d.GetOk("hcl2")
```

Comment:
That's right, but I kind of did that on purpose to avoid changes to state as much as possible. If you're not using HCL2, you probably have a reason for that, so switching to it outside of Terraform means you're doing something funky 😛 

But if that's surprising then https://github.com/hashicorp/terraform-provider-nomad/pull/405/commits/629174e9b9d1315ace70f105f665383e134ae474 forces updates if variables change. Is this better?